CMakeLists.txt

```diff
@@ -215,26 +215,38 @@ if (NOT HAVE_WORKING_WERROR)
 endif()
 
 # Use Python modules based on CMake version for backward compatibility
-set(CYTHON_EXECUTABLE "")
 if (${CMAKE_VERSION} VERSION_LESS "3.12")
-  # Look for Python. We prefer some variant of python 3 if the system has it
-  FIND_PACKAGE(PythonInterp 3 QUIET)
-  if (PythonInterp_FOUND)
-    # pyverbs can only use python3:
-    if (NOT NO_PYVERBS)
-      FIND_PACKAGE(cython)
-    endif()
-  else()
-    # But we still must have python (be it 2) for the build process
-    FIND_PACKAGE(PythonInterp REQUIRED)
-  endif()
-else()
-  # FindPython looks preferably for Python3. If not found, version 2 is searched
-  FIND_PACKAGE(Python COMPONENTS Interpreter REQUIRED)
-  set(PYTHON_EXECUTABLE ${Python_EXECUTABLE})
-  if (NOT NO_PYVERBS AND Python_VERSION_MAJOR EQUAL 3)
+        FIND_PACKAGE(PythonInterp REQUIRED)
+        FIND_PACKAGE(PythonLibs ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR} EXACT)
+elseif (${CMAKE_VERSION} VERSION_GREATER "3.12")
+        set(Python_EXECUTABLE ${PYTHON_EXECUTABLE})
+        FIND_PACKAGE(Python 3 REQUIRED COMPONENTS Interpreter OPTIONAL_COMPONENTS Development)
```

Comment:
What fails cmake if cython is enabled and we don't find Development?

Reply:
If find_package python find developer libs Python_Development_FOUND variable is set so i check for that